source/agora/consensus/state/ValidatorSet.d

```diff
@@ -94,15 +95,20 @@ public class ValidatorSet
         if (this.hasEnrollment(enroll.utxo_key))
             return "This validator is already enrolled";
 
+        // check if an enrollment of the same public key is already present
+        if (this.hasPublicKey(pubkey))
```

Comment:
The meaning of `add` being called is that it's been already approved by the network. So calling this function here is not proper. The checking code should move into the `isInvalidCandidateReason` function of `EnrollmentManager`. I will fix it.